components/gantry/oneaxis/oneaxis.go

```diff
@@ -187,10 +189,79 @@ func (g *oneAxis) Reconfigure(ctx context.Context, deps resource.Dependencies, c
 			return err
 		}
 	}
+	ctx, cancelFunc := context.WithCancel(context.Background())
+	g.cancelFunc = cancelFunc
+	g.checkHit(ctx)
 
 	return nil
 }
 
+func (g *oneAxis) checkHit(ctx context.Context) {
+	g.activeBackgroundWorkers.Add(1)
+	utils.PanicCapturingGo(func() {
+		defer utils.UncheckedErrorFunc(func() error {
+			return g.motor.Stop(ctx, nil)
+		})
+		defer g.activeBackgroundWorkers.Done()
+		for {
+			if ctx.Err() != nil {
+				return
+			}
+			select {
+			case <-ctx.Done():
+				return
+			default:
+			}
+			func() {
+				for i := 0; i < len(g.limitSwitchPins); i++ {
+					hit, err := g.limitHit(ctx, i)
+					if err != nil {
+						g.logger.Error(err)
+						return
+					}
+					if hit {
+						if err := g.moveAway(ctx, i); err != nil {
+							g.logger.Error(err)
+							return
+						}
+					}
+				}
+			}()
+		}
+	})
+}
+
+// Once a limit switch is hit in any move call (from the motor or the gantry component),
+// this function stops the motor, and reverses the direction of movement until the limit
+// switch is no longer activated.
+func (g *oneAxis) moveAway(ctx context.Context, pin int) error {
+	if err := g.motor.Stop(ctx, nil); err != nil {
+		return err
+	}
+	time.Sleep(100 * time.Millisecond)
+	if pin == 0 {
+		if err := g.motor.GoFor(ctx, g.rpm, 0, nil); err != nil {
+			return err
+		}
+	} else {
+		if err := g.motor.GoFor(ctx, -1.0*g.rpm, 0, nil); err != nil {
+			return err
+		}
+	}
+	for {
+		hit, err := g.limitHit(ctx, pin)
+		if err != nil {
+			return err
+		}
+		if !hit {
+			if err := g.motor.Stop(ctx, nil); err != nil {
+				g.logger.Error(err)
```

Comment:
Every other error in this function gets returned, rather than logged. Why not this one?